From cbe045cb6a8126c5be9062caf039ada20adc5480 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Thu, 7 Jan 2016 13:53:39 +0100
Subject: [PATCH] build with debug info

---
 hack/make/binary | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hack/make/binary b/hack/make/binary
index 5ddce8d..75447a8 100644
--- a/hack/make/binary
+++ b/hack/make/binary
@@ -40,7 +40,7 @@ echo "Building: $DEST/$BINARY_FULLNAME"
 go build \
 	-o "$DEST/$BINARY_FULLNAME" \
 	"${BUILDFLAGS[@]}" \
-	-ldflags "
+	-ldflags "-B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \n')
 		$LDFLAGS
 		$LDFLAGS_STATIC_DOCKER
 	" \
-- 
1.9.3

