Index: docker-6ec29ef9a2d48a6ccd716a2bbb00a1ed07412d6a/api/server/credentials_linux.go
===================================================================
--- docker-6ec29ef9a2d48a6ccd716a2bbb00a1ed07412d6a.orig/api/server/credentials_linux.go
+++ docker-6ec29ef9a2d48a6ccd716a2bbb00a1ed07412d6a/api/server/credentials_linux.go
@@ -213,7 +213,7 @@ func (s *Server) LogAction(w http.Respon
 			break
 		}
 		message = fmt.Sprintf("LoginUID=%v, %s", loginuid, message)
-		if loginuid == 0xffffffff { // -1 means no login user
+		if uint32(loginuid) == 0xffffffff { // -1 means no login user
 			//No login UID is set, so no point in looking up a name
 			break
 		}
