diff --git a/hack/make/dynbinary b/hack/make/dynbinary
index e5fc0be..a8ec909 100644
--- a/hack/make/dynbinary
+++ b/hack/make/dynbinary
@@ -13,7 +13,7 @@ fi
 
 (
 	export IAMSTATIC="false"
-	export LDFLAGS_STATIC_DOCKER=''
+	export LDFLAGS_STATIC_DOCKER='-linkmode external -extldflags -zmuldefs'
 	export BUILDFLAGS=( "${BUILDFLAGS[@]/netgo /}" ) # disable netgo, since we don't need it for a dynamic binary
 	export BUILDFLAGS=( "${BUILDFLAGS[@]/static_build /}" ) # we're not building a "static" binary here
 	source "${MAKEDIR}/binary"
